src/flows/shared/steps/OrganizeCollection/Sidebar/Sidebar.tsx

```diff
@@ -196,17 +192,62 @@ const SidebarTokens = ({ nftFragmentsKeyedByID, tokens }: SidebarTokensProps) =>
     mount();
   }, [nftFragmentsKeyedByID, reportError, tokens]);
 
-  return (
-    <>
-      {displayedTokens.map((editModeToken) => (
+  /**
+   * We render a row with three token.
+   */
+  const rowRenderer = ({ key, style, index }: ListRowProps) => {
+    const items = [];
+
+    // add blank block button at the beginning of list
+    if (index === 0) {
+      items.push(
+        <StyledAddBlankBlock onClick={handleAddBlankBlockClick}>
+          <StyledAddBlankBlockText>Add Blank Space</StyledAddBlankBlockText>
+        </StyledAddBlankBlock>
+      );
+    }
+
+    const fromIndex = index === 0 ? 0 : index * COLUMN_COUNT - 1;
```

Comment:
This part is kind of hacky. Would love to know any thought if its could be better

Reply:
no problem :) unfortunately this is the result of our hacky setup with the whitespace adder

Reply:
hmm maybe we could have a memoized value thats like `const rows = useMemo()`, which would memoize the list of rows, including the placeholder. 
Then rowRenderer doesn't have to compute the rows every time, or think about the whitespace block and handle any row index specially. 

Thinking the memoized value would splice a blank block placeholder to the beginning of `displayedTokens`, so that it looks like this `['placeholder', token1, token2, etc]`
and then it would convert it into an an array of rows: `[[row1 items including placeholder], [row2items]]`

and then rowRenderer could use this list instead of `displayedTokens` ?
just an idea

Reply:
That's interesting idea @kaitoo1 . Since its expensive func, it make sense to memoized it + 2d array. Will try this idea!

Reply:
Updated it with memoized rows! It doesnt seems so hacky like before 😬

Reply:
nice!! that's great

Reply:
awesome, great idea @kaitoo1!!